buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -507,8 +502,8 @@ private long collapsePast(long handle) {
             int runOffset = runOffset(handle);
             int runPages = runPages(handle);
 
-            Long pastRun = getAvailRunByOffset(runOffset - 1);
-            if (pastRun == null) {
+            long pastRun = getAvailRunByOffset(runOffset - 1);
+            if (pastRun == 0) {
```

Comment:
```suggestion
            if (pastRun == -1) {
```

== 0 seems incorrect